core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -157,11 +161,7 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
     };
 
     Topic.prototype.addAdditionalStoryId = function(storyId) {
-      if (this._additionalStoryIds.indexOf(storyId) === -1) {
-        this._additionalStoryIds.push(storyId);
-        return true;
-      }
-      return false;
+      this._additionalStoryIds.push(storyId);
```

Comment:
Here, we don't have a check for whether the id exists beforehand; in addUncategorizedSkillId, we do. What is the rationale for doing this in some places and not in others?

Reply:
For these, I added the check in TopicUpdateService, while for moving skills, the check is in TopicObjectFactory. Shall I have all checks in TopicUpdateService.js then?

Reply:
As long as it's consistent, I don't have a strong opinion. So let's do it all in one place or all in another place, but not a mixture of both (unless there's a reason to do the latter, which should be clearly enunciated).

Reply:
Done, moved all such checks to TopicUpdateService, since that's what the external controllers would be calling and so any errors can be found there itself, before the changelists are modified.